cfme/utils/ssh.py

```diff
@@ -842,3 +849,35 @@ def keygen():
     with _ssh_pubkey_file.open('w') as f:
         f.write("{} {} {}\n".format(pub.get_name(), pub.get_base64(),
             'autogenerated cfme_tests key'))
+
+
+def unquirked_ssh_client(hostname, *args, **kwargs):
+    """ A factory for the SSHClient dealing with some of its quirks. """
+    if hostname is None:
+        raise ValueError(f'Hostname cannot be {hostname}')
```

Comment:
The f-string seems unnecessary, since you already know hostname is None. Maybe also check if it's an empty string, e.g.,
```
if not hostname:
    raise ValueError('No hostname specified.')

Reply:
Oh!